docs-website/src/components/FetchFileCodeBlock.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import CodeBlock from "@theme/CodeBlock";
+
+const FetchFileCodeBlock = ({ src, component, raw, showLink = true, fileFormat, fileName }) => {
+  const [content, setContent] = useState(null);
+
+  const linkBaseUrl = `${src}`.replace(/^\/build\/\.\.\//gm, '').replace(/\/zarf\.yaml.+?$/gm, '');
+
+  const handleDownloadClick = () => {
```

Comment:
This seems round about - can you set `download` on the a tag below with the filename (using the data uri if it is json?)

Reply:
Also JS has `.replaceAll` if you so choose to use it

Reply:
thanks for the comments, like put it all in the a component?